AxonDeepSeg/download_model.py

```diff
@@ -14,9 +15,17 @@ def download_model(destination = None):
         tem_destination = destination / "default_TEM_model"
         model_seg_pns_bf_destination = destination / "model_seg_pns_bf"
 
-    url_TEM_model = "https://osf.io/2hcfv/?action=download&version=5"  # URL of TEM model hosted on OSF storage with the appropriate versioning on OSF
-    url_SEM_model = "https://osf.io/sv7u2/?action=download&version=5"  # URL of SEM model hosted on OSF storage with the appropriate versioning on OSF
-    url_model_seg_pns_bf = "https://osf.io/hfeu2/?action=download&version=2" # URL of Bright Field Optical Microscopy model hosted on storage with the appropriate versioning on OSF
+    # retrieve latest release identifier from github API
+    response_TEM = requests.get("https://api.github.com/repos/axondeepseg/default-TEM-model/releases/latest")
+    folder_name_TEM_model = Path("default-TEM-model-"+str(response_TEM.json()['tag_name']))
+    response_SEM = requests.get("https://api.github.com/repos/axondeepseg/default-SEM-model/releases/latest")
+    folder_name_SEM_model = Path("default-SEM-model-"+str(response_SEM.json()['tag_name']))
+    response_OM = requests.get("https://api.github.com/repos/axondeepseg/model-seg-pns-bf/releases/latest")
+    folder_name_OM_model = Path("model-seg-pns-bf-"+str(response_OM.json()['tag_name']))
+
+    url_TEM_model = "https://github.com/axondeepseg/default-TEM-model/archive/refs/tags/"+str(response_TEM.json()['tag_name'])+".zip" 
+    url_SEM_model = "https://github.com/axondeepseg/default-SEM-model/archive/refs/tags/"+str(response_SEM.json()['tag_name'])+".zip" 
+    url_model_seg_pns_bf = "https://github.com/axondeepseg/model-seg-pns-bf/archive/refs/tags/"+str(response_OM.json()['tag_name'])+".zip"
```

Comment:
I was thinking that perhaps we should store the url in [config.py](https://github.com/neuropoly/axondeepseg/blob/master/config.py) script rather than hardcoing in this script. By doing so, all the hardcoded strings will be centralized in a single location. What do you think?

Reply:
That makes a lot of sense. I will incorporate this change right away!

Reply:
In fact I'll wait to close our discussion above (concerning the actual URLs we want to keep) before commiting these changes.